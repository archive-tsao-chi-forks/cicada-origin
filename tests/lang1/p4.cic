{
  one : Nat
  one = add1(zero)

  two : Nat
  two = add1(one)

  id : (Nat) -> Nat
  id = (x) => x

  (_) => (_) => id(id(id(id(two))))

} : (Nat) -> (Nat) -> Nat
