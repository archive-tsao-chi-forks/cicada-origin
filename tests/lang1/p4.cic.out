(_1) => (_2) => 2: (Nat) -> (Nat) -> Nat
